docs/README.md

```diff
@@ -1,41 +1,42 @@
+## Usage in adidas
+MVI library is now being used in adidas CONFIRMED app. In fact, this library was developed initially as core architecture of CONFIRMED app and later it was decided to open source it.
 
-### Build
+## Installation
+To use the ADIDAS MVI Library in your Android project, follow these steps:
 
+Open the build.gradle file of your project.
+Add the following dependency to the dependencies block:
 ```
-$ yarn build
+implementation("com.adidas.mvi:mvi:{LATEST_VERSION}")
 ```
+Sync your project with the Gradle files.
+## Usage
+To use the ADIDAS MVI Library in your application, you need to understand the basic concepts and components of the MVI pattern.
 
-This command generates static content into the `build` directory and can be served using any static contents hosting service.
-
-### Deployment
-
-Using SSH:
-
-```
-$ USE_SSH=true yarn deploy
-```
-
-Not using SSH:
-
-```
-$ GIT_USER=<Your GitHub username> yarn deploy
-```
+## Concepts
+Model: The model represents the state of your application. It's an immutable data structure that holds all the information needed to render the UI.
+View: The view is responsible for rendering the UI and reacting to user interactions. It observes the model and updates the UI accordingly.
+Intent: An intent represents a user action or an event that triggers a state change in the model. Intents are dispatched to the model, which updates its state based on the intent received.
+## Components
+ADIDAS MVI Library provides the following components to help you implement the MVI pattern:
 
-If you are using GitHub pages for hosting, this command is a convenient way to build the website and push to the `gh-pages` branch.
+**View State**: Represents the immutable model data that the view observes. It holds all the necessary information required to render the user interface (UI). The ViewState is updated by the model based on the received intents and reflects the current state of the application.
+**Side Effect**: Represents an action or an event that occurs as a result of the state update. It encapsulates tasks such as network requests, database operations, or other external interactions. Side effects are triggered by the model and can be used to perform background operations or trigger additional state changes. This action needs to be executed only once.
+**State**: The State<ViewState, SideEffect> class combines the ViewState and SideEffect components to provide a comprehensive representation of the application state within the MVI pattern.
+**State Transform**: This interface helps with transforming from one state to another, working closely with Reducer, it encapsulates the necessary information for the model to update its state. There are some helper functions such as requireAndReduceState() for more convinient.
+**Reducer**: Serves as a middleware or intermediary component between the all MVI concepts and UI.
```

Comment:
Reducer is also responsible of running the intents with a specific coroutine dispatcher, cancelation and logging of events.